tree/ntuple/v7/src/RField.cxx

```diff
@@ -1083,6 +1083,211 @@ void ROOT::Experimental::RVectorField::AcceptVisitor(Detail::RFieldVisitor &visi
+{
+   EnsureColumnType({EColumnType::kIndex}, 0, desc);
+   GenerateColumnsImpl();
+}
+
+ROOT::Experimental::Detail::RFieldValue ROOT::Experimental::RRVecField::GenerateValue(void* where)
+{
+   // initialize data members fBegin, fSize, fCapacity
+   // currently the inline buffer is left uninitialized
+   void **beginPtr = new (where) (void *)(nullptr);
+   int32_t* sizePtr = new (reinterpret_cast<void*>(beginPtr + 1)) int32_t(0);
+   new (sizePtr + 1) int32_t(0);
+
+   return Detail::RFieldValue(/*captureTag*/true, this, where);
+}
+
+void ROOT::Experimental::RRVecField::DestroyValue(const Detail::RFieldValue& value, bool dtorOnly)
+{
+   void **rvecAddr = reinterpret_cast<void**>(value.GetRawPtr());
+   // void* fBegin is the first data member
+   void *begin = *rvecAddr;
+   // int32_t fSize is the second data member (after 1 void*, no padding expected)
+   const int32_t* sizePtr = reinterpret_cast<int32_t *>(rvecAddr + 1);
+   const int32_t size = *sizePtr;
+   R__ASSERT(size >= 0);
+   // int32_t fCapacity is the third data member (after fSize, no padding expected)
+   const int32_t capacity = *(sizePtr + 1);
+
+   for (int32_t i = 0; i < size; ++i) {
+      auto elementValue = fSubFields[0]->CaptureValue(reinterpret_cast<void *>(std::intptr_t(begin) + i * fItemSize));
+      fSubFields[0]->DestroyValue(elementValue, true /* dtorOnly */);
+   }
+
+   // figure out if we are in the small state, i.e. begin == &inlineBuffer
+   // there might be padding between fCapacity and the inline buffer, so we compute it here
+   const auto dataMemberSz = sizeof(void *) + 2 * sizeof(int32_t);
+   const auto alignOfT = fSubFields[0]->GetAlignment();
+   auto paddingMiddle = dataMemberSz % alignOfT;
+   if (paddingMiddle != 0)
+      paddingMiddle = alignOfT - paddingMiddle;
+   const bool isSmall = begin == (rvecAddr + dataMemberSz + paddingMiddle);
+
+   const bool owns = capacity != -1;
+   if (!dtorOnly && !isSmall && owns)
+      free(begin);
+}
+
+ROOT::Experimental::Detail::RFieldValue ROOT::Experimental::RRVecField::CaptureValue(void* where)
+{
+   return Detail::RFieldValue(true /* captureFlag */, this, where);
+}
+
+std::vector<ROOT::Experimental::Detail::RFieldValue>
+ROOT::Experimental::RRVecField::SplitValue(const Detail::RFieldValue &value) const
+{
+   void **rvecAddr = reinterpret_cast<void**>(value.GetRawPtr());
+   // void* fBegin is the first data member
+   void *begin = *rvecAddr;
+   // int32_t fSize is the second data member (after 1 void*)
+   const int32_t size = *reinterpret_cast<int32_t *>(rvecAddr + 1);
+   R__ASSERT(size >= 0);
+
+   std::vector<Detail::RFieldValue> result;
+   for (int32_t i = 0; i < size; ++i) {
+      auto elementValue = fSubFields[0]->CaptureValue(reinterpret_cast<void *>(std::intptr_t(begin) + i * fItemSize));
+      result.emplace_back(std::move(elementValue));
+   }
+   return result;
+}
+
+size_t ROOT::Experimental::RRVecField::GetValueSize() const
```

Comment:
It might make sense to cache the size.

Reply:
Should we evaluate it once in the ctor or evaluate it lazily and cache it?

Reply:
(at construction time)